### ISSUE TITLE
Query: Undisposed DbDataReader objects

### ISSUE THREAD

**Issue by bricelam** [aspnet/EntityFrameworkCore#9181]
The following tests are not disposing `DbDataReader` objects leaving them to be collected by the GC.

* src\EFCore.Relational.Specification.Tests\Query\FromSqlQueryTestBase.cs
  * Bad_data_error_handling_null
  * Bad_data_error_handling_null_no_tracking
  * Bad_data_error_handling_null_projection
  * From_sql_queryable_simple_columns_out_of_order_and_not_enough_columns_throws
* src\EFCore.Specification.Tests\Query\QueryTestBase.ResultOperators.cs
  * Average_no_data_subquery
  * Max_no_data_subquery
  * Min_no_data_subquery

In case it helps, here is the stack trace of the objects' creations.
```
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues) in src\EFCore.Relational\Storage\Internal\RelationalCommand.cs:line 118
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(Boolean buffer) in src\EFCore.Relational\Query\Internal\QueryingEnumerable.cs:line 102
   at Microsoft.EntityFrameworkCore.ExecutionStrategyExtensions.<>c__DisplayClass12_0`2.<Execute>b__0(DbContext c, TState s) in src\EFCore\Extensions\ExecutionStrategyExtensions.cs:line 331
   at Microsoft.EntityFrameworkCore.Storage.Internal.NoopExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded) in src\EFCore\Storage\Internal\NoopExecutionStrategy.cs:line 39
   at Microsoft.EntityFrameworkCore.ExecutionStrategyExtensions.Execute[TState,TResult](IExecutionStrategy strategy, Func`2 operation, Func`2 verifySucceeded, TState state) in src\EFCore\Extensions\ExecutionStrategyExtensions.cs:line 329
   at Microsoft.EntityFrameworkCore.ExecutionStrategyExtensions.Execute[TState,TResult](IExecutionStrategy strategy, TState state, Func`2 operation) in src\EFCore\Extensions\ExecutionStrategyExtensions.cs:line 279
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext() in src\EFCore.Relational\Query\Internal\QueryingEnumerable.cs:line 75
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext() in src\EFCore\Query\Internal\LinqOperatorProvider.cs:line 131
```

I tried investigating more, but started drowning in expression trees...

**Comment by ajcvickers** [comment]
@anpete High priority, but not higher than that sample provider thing.

**Comment by anpete** [comment]
@smitpatel Done? Dupe?

**Comment by bricelam** [comment]
@AndriySvyryd may have fixed this already too...

**Comment by bricelam** [comment]
Possibly by #9454

**Comment by smitpatel** [comment]
Verified that all above tests are fixed in #9454
